models/stock_move_line.py

```diff
@@ -426,3 +424,43 @@ def get_info(self):
             res.append(line._prepare_info())
 
         return res
+
+    def _prepare_task_info(self):
+        """ Prepares info of a task
+        """
+        Quant = self.env['stock.quant']
+
+        self.ensure_one()
+
+        task = {
+            'picking_id': self.picking_id.id,
+        }
+        user_scans = self.picking_id.picking_type_id.u_user_scans
+        if user_scans == 'product':
+            task['pick_quantity'] = self.product_qty
+            quant = Quant._gather(self.product_id, self.location_id,
+                                  lot_id=self.lot_id, package_id=self.package_id,
+                                  owner_id=self.owner_id, strict=True)
+            task['quant_id'] = quant.get_info()[0]
+        else:
+            task['package_id'] = self.package_id.get_info(extended=True)[0]
+
+        return task
+
+    def sort_by_location_product(self, state=None):
+        """ Sort the move lines
+        """
+        MoveLine = self.env['stock.move.line']
+
+        mls = self
+
+        if state is None:
+            pass
+        elif state == 'done':
+            mls = mls.filtered(lambda ml: ml.qty_done == ml.product_qty)
+        elif state == 'not_done':
+            mls = mls.filtered(lambda ml: ml.qty_done < ml.product_qty)
+        else:
+            assert False, 'State not valid to sort tasks'
```

Comment:
shouldn't this be raise error() instead of assert?

Reply:
Semantically they are different. If we get here is a coding error, not a system error that is why we have used an assert.

Reply:
Yes but as assertions can be turned off it wouldn't be safer to raise an AssertionError